docs/reference/files/file-charmcraft-yaml.rst

```diff
@@ -0,0 +1,901 @@
+.. _file-charmcraft-yaml:
+.. highlight:: yaml
```

Comment:
Since `highlight` acts as a switch, I think the code would be easier to read if it were placed immediately before the first block that uses it.

Reply:
That's pretty unintuitive to me - if I'm looking for why a block is highlighted in an unexpected manner, I'm going to check that block (which is likely not the first block in the page, especially for reference pages like this), then the directives at the very top of the page, then the global `highlight_language` configuration.

A `highlight` directive somewhere in the middle of a file would be highly unappreciated, much like an `import` coming between two class definitions in a source file. Worse still would be a switch of the `highlight` directive mid-file, which would be a form of code smell that this document is about two different things (and thus should be broken up into multiple pages), since in cases where one needs to change the highlight language a `code-block` can set its own language as a one-off.